crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -472,6 +479,17 @@ async fn payment_response_update_tracker<F: Clone, T: types::Capturable>(
                         flow_name.clone(),
                     )
                     .await;
+
+                    let additional_payment_method_data =
+                        update_additional_payment_data_with_connector_response_pm_data(
+                            payment_data.payment_attempt.payment_method_data.clone(),
+                            router_data
+                                .connector_response
+                                .and_then(|connector_response| {
+                                    connector_response.additional_payment_method_data
+                                }),
+                        )?;
+
```

Comment:
I think this can be done outside the match statement and `additional_payment_method_data` can be reused inside the match arms.